File: Datalake/Datalake/Models/HistoryResponse.cs
using Datalake.Enums;
using System.Collections.Generic;

namespace Datalake.Models
{
	public class HistoryResponse
	{
		public int Id { get; set; }

		public string TagName { get; set; }

		public TagType Type { get; set; }

		public AggFunc Func { get; set; }

		public List<HistoryValue> Values { get; set; } = new List<HistoryValue>();
	}
}

File: Datalake/datalakeclient/src/@types/HistoryResponse.ts
import { HistoryValue } from "./HistoryValue"
import { AggFunc } from "./enums/AggFunc"
import { TagType } from "./enums/TagType"

export interface HistoryResponse {
	Id: number
	TagName: string
	Type: TagType
	Func: AggFunc
	Values: HistoryValue[]
}